src/test/java/org/jabref/gui/search/GlobalSearchBarTest.java

```diff
@@ -0,0 +1,156 @@
+ * @version 1.0
+ * @since May 2023
+ */
+
+@GUITest
+@ExtendWith(ApplicationExtension.class)
+public class GlobalSearchBarTest {
+    private Stage stage;
+    private Scene scene;
+    private HBox hBox;
+
+    private GlobalSearchBar searchBar;
+    private StateManager stateManager;
+
+    /**
+     * The onStart method, accepts the stage of the application and sets
+     * up the basic components of the UI (i.e. scene, layout), that will
+     * be used for our tests. Also interacts with GlobalSearchBar and
+     * StateManager classes, in order to set up the search box.
+     *
+     * @param stage Reference to the Stage class, part of the JavaFX UI toolkit
+     */
+    @Start
+    public void onStart(Stage stage) {
+        SearchPreferences searchPreferences = mock(SearchPreferences.class);
+        when(searchPreferences.getSearchFlags()).thenReturn(EnumSet.noneOf(SearchRules.SearchFlags.class));
+        PreferencesService prefs = mock(PreferencesService.class, Answers.RETURNS_DEEP_STUBS);
+        when(prefs.getSearchPreferences()).thenReturn(searchPreferences);
+
+        stateManager = new StateManager();
+        // need for active database, otherwise the searchField will be disabled
+        stateManager.setActiveDatabase(new BibDatabaseContext());
+
+        // instantiate GlobalSearchBar class, so the change listener is registered
+        searchBar = new GlobalSearchBar(
+                mock(JabRefFrame.class),
+                stateManager,
+                prefs,
+                mock(CountingUndoManager.class),
+                mock(DialogService.class)
+        );
+
+        hBox = new HBox(searchBar);
+
+        scene = new Scene(hBox, 400, 400);
+        this.stage = stage;
+        stage.setScene(scene);
+
+        stage.show();
+    }
+
+    /**
+     * The recordingSearchQueriesOnFocusLost method, accepts a robot reference
+     * to the FxRobot class and verifies that unnecessary prefixes of the
+     * search query are not recorded. Essentially, the query is recorded
+     * only when the searchField node loses focus.
+     *
+     * @param robot Reference to the FxRobot class, a TestFx utility class
+     */
+
+    @Test
+    void recordingSearchQueriesOnFocusLost(FxRobot robot) {
+        stateManager.clearSearchHistory();
+        String searchQuery = "Smith";
+        // track the node, that the search query will be typed into
+        TextInputControl searchField = robot.lookup("#searchField").queryTextInputControl();
+
+        // the focus is on searchField node, as we click on the search box
+        var searchFieldRoboto = robot.clickOn(searchField);
+        for (char c : searchQuery.toCharArray()) {
+            searchFieldRoboto.write(String.valueOf(c));
+            try {
+                Thread.sleep(401);
+                Assertions.assertTrue(stateManager.getWholeSearchHistory().isEmpty());
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
```

Comment:
No need for try-catch in Test cases. Just add ` throws Exception` to the method signature.

I don't think, that `InterruptedException` should be ignored and then the test continue.